### DISCUSSION TITLE
Upload again after error occurred

### DISCUSSION THREAD

**Discussion by masterwishx** [simulot/immich-go#436] [Q&A] [answered]
Thanks for amazing tool.
Using lasted version. 
When runned upload 3 takeout zip files
Some error occurred at 90%.
If I run immich-go again it will continue from where finished or will start from begin?

**Comment by simulot** [comment, marked as answer]
You can re-launch the same command.
Immich-go skips files that are already uploaded.

**Reply by masterwishx** [reply]
> You can re-launch the same command. Immich-go skips files that are already uploaded.

Thanks got it

**Reply by masterwishx** [reply]
> You can re-launch the same command. Immich-go skips files that are already uploaded.

i have still error near the end atfer some times tryed : 

![image](https://github.com/user-attachments/assets/70579d0a-d629-4949-a9b8-1cc15c0ca8f3)

**Reply by simulot** [reply]
What the log files says?

**Reply by masterwishx** [reply]
> What the log files says?

this is the log file end , i runned now with -api-trace ... 

when runned 2 times after first time upload added 11 files every time

**Reply by masterwishx** [reply]
this is when tryed with -api-trace and -upload-when-missing-JSON in case i have this without json :  

```
2024-08-16T20:03:38+03:00 QUERY 8245 GetJobs GET http://192.168.0.199:8180/api/jobs
   Accept [application/json]
   X-Api-Key redacted
-- Empty body or binary body not dumped --
2024-08-16T20:03:38+03:00 RESPONSE 8245 GetJobs GET http://192.168.0.199:8180/api/jobs
  Status: 200 OK
-- response body --
{
 "backgroundTask": {
  "jobCounts": {
   "active": 0,
   "completed": 0,
   "failed": 0,
   "delayed": 0,
   "waiting": 0,
   "paused": 0
  },
  "queueStatus": {
   "isActive": false,
   "isPaused": false
  }
 },
 "duplicateDetection": {
  "jobCounts": {
   "active": 0,
   "completed": 0,
   "failed": 0,
   "delayed": 0,
   "waiting": 0,
   "paused": 0
  },
  "queueStatus": {
   "isActive": false,
   "isPaused": false
  }
 },
 "faceDetection": {
  "jobCounts": {
   "active": 0,
   "completed": 0,
   "failed": 0,
   "delayed": 0,
   "waiting": 0,
   "paused": 0
  },
  "queueStatus": {
   "isActive": false,
   "isPaused": false
  }
 },
 "facialRecognition": {
  "jobCounts": {
   "active": 0,
   "completed": 0,
   "failed": 0,
   "delayed": 0,
   "waiting": 0,
   "paused": 0
  },
  "queueStatus": {
   "isActive": false,
   "isPaused": false
  }
 },
 "library": {
  "jobCounts": {
   "active": 0,
   "completed": 0,
   "failed": 0,
   "delayed": 0,
   "waiting": 0,
   "paused": 0
  },
  "queueStatus": {
   "isActive": false,
   "isPaused": false
  }
 },
 "metadataExtraction": {
  "jobCounts": {
   "active": 0,
   "completed": 0,
   "failed": 0,
   "delayed": 0,
   "waiting": 0,
   "paused": 0
  },
  "queueStatus": {
   "isActive": false,
   "isPaused": false
  }
 },
 "migration": {
  "jobCounts": {
   "active": 0,
   "completed": 0,
   "failed": 0,
   "delayed": 0,
   "waiting": 0,
   "paused": 0
  },
  "queueStatus": {
   "isActive": false,
   "isPaused": false
  }
 },
 "notifications": {
  "jobCounts": {
.... truncated ....
-- response body end --


```

**Reply by simulot** [reply]
hum... apparently the upload fails... but it's not correctly reported... As concerned files are not on the server, immich-go tries again.

Could you zip the log file and DM it on my discord @simulot  ?

**Reply by masterwishx** [reply]
also have this on server: 
![image](https://github.com/user-attachments/assets/68f1f4df-7093-479f-bf47-5cad0087472c)

**Reply by simulot** [reply]
the `-api-trace` generates huge files... "Getjobs" is called several times per second to display the server's activity

**Reply by masterwishx** [reply]
> the `-api-trace` generates huge files... "Getjobs" is called several times per second to display the server's activity

sorry i meaned this is log from server Immich using [ghcr.io/imagegenius/immich](https://github.com/orgs/imagegenius/packages/container/package/immich)

**Reply by simulot** [reply]
I'd don't recognize immich's log

**Comment by Cherianpaul** [comment]
Hi 
   It is a nice tool which helps to upload the google takeout files to my Immich server.
   **But one suggestion or idea** : 
   Each time when an error occurs due to server timeout or file error after completing 90%, it checks each file with server for integrity and skips them if present. For this we need to wait a long time. 
   Instead _can't we provide an option to save the successfully uploaded file history in a txt or json file_ so that the tool can check _locally_ inside this history file _and continue from where it stopped last time_?
   Once the upload completed successfully, the history file can be deleted.
   This feature can be optional if the user prefer to do so. By default it can do the integrity check like now. 
   it will be a great help so that we do not have to wait until it skips and reaches 90% again as google takeout files are more in numbers like 1TB.
Thanks and Regards
Cherian Paul

**Reply by simulot** [reply]
I don't think this is possible to resume the process after a crash. The actions depend on the server content that may have changed in the meantime. Example: immich doesn't like at all uploading duplicated files. Immich-go takes the list of server files at start-up, than look at the input and determine which files need to be uploaded. What is files have been uploaded on the server by another mean? 

This said, there are some possibilities:
- the [v0.24.7](https://github.com/simulot/immich-go/releases/tag/v0.24.7) is better regarding errors. It should have much less errors 
- you can force immicg-go to continue afer errors: `--on-server-errors=CONTINUE`

**Reply by Cherianpaul** [reply]
Thanks for the quick reply. 
I was using the latest v0.24.2 (latest one now) 
But I will try these 2 options with pre-built [v0.24.7](https://github.com/simulot/immich-go/releases/tag/v0.24.7) as a work around for my problem.
